File: ios/TencentAd.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(TencentAd, NSObject)

RCT_EXTERN_METHOD(registerAppId:(NSString)appId
                  resolve:(RCTPromiseResolveBlock)resolve
                  reject:(RCTPromiseRejectBlock)reject)

@end

File: ios/TencentAd.swift
@objc(TencentAd)
class TencentAd: NSObject {
    @objc(registerAppId:resolve:reject:)
    func registerAppId(appId: String, resolve: RCTPromiseResolveBlock, reject: RCTPromiseRejectBlock) -> Void {
        let result: Bool = GDTSDKConfig.registerAppId(appId);
        if result {
            resolve("")
        } else {
            reject("-1", "", nil)
        }
    }
}
